package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -315,6 +329,9 @@ def __init__(
             ]
             for t in heavyatom_names
         }
+        self._donor_mask: Optional[np.ndarray] = (
+            ag.residues.resnames != "PRO" if ignore_proline_donor else None
+        )
```

Comment:
This may not be correct. The code runs ... but I am not sure if I should be masking corresponding atoms.

Reply:
if the question is about shape of the donor mask (i.e. make it for residues or for atoms), then it seems to be correct -- pydssp makes it of shape `l` ([line](https://github.com/ShintaroMinami/PyDSSP/blob/e251a43ff8622fe0a555313b1567edce45e789e8/pydssp/pydssp_numpy.py#L66)) and `l` is the number of residues ([link](https://github.com/ShintaroMinami/PyDSSP/blob/e251a43ff8622fe0a555313b1567edce45e789e8/pydssp/pydssp_numpy.py#L45)). However, the notation I made in `pydssp_numpy.py` is wrong, hence I renamed it in the commit below.